packages/playwright-core/src/utils/fileUtils.ts

```diff
@@ -25,12 +24,10 @@ export async function mkdirIfNeeded(filePath: string) {
   await fs.promises.mkdir(path.dirname(filePath), { recursive: true }).catch(() => {});
 }
 
-export async function removeFolders(dirs: string[]): Promise<Array<Error|null|undefined>> {
-  return await Promise.all(dirs.map((dir: string) => {
-    return new Promise<Error|null|undefined>(fulfill => {
-      rimraf(dir, { maxRetries: 10 }).then(() => fulfill(undefined)).catch((e: Error) => fulfill(e));
-    });
-  }));
+export async function removeFolders(dirs: string[]): Promise<Error[]> {
+  return await Promise.all(dirs.map((dir: string) =>
+    fs.promises.rm(dir, { recursive: true, force: true, maxRetries: 10 })
+  )).catch(e => e);
```

Comment:
This looks like a mistake of `.catch(e => e)));`, i.e. should catch each error of `fs.promises.rm`. Otherwise the return type should be `Promise<Error>`.

Reply:
```
const fs = require('fs');
const dirs = ['/tmp/x', '/tmp/y'];

(async () => {
  await Promise.all(dirs.map((dir) => Promise.reject({ dir }))).catch(e => e).then(x => console.log(x));
  // { dir: '/tmp/x' }

  await Promise.all(dirs.map((dir) => Promise.reject({ dir }).catch(e => e))).then(x => console.log(x));
  // [ { dir: '/tmp/x' }, { dir: '/tmp/y' } ]
})();
```

Reply:
As a result, the following line of `packages/playwright/src/runner/tasks.ts` yields `TypeError: object is not iterable (cannot read property Symbol(Symbol.iterator))`.
```
await Promise.all(Array.from(outputDirs).map(outputDir => removeFolders([outputDir]).then(async ([error]) => {
```

Reply:
Good catch, are you interested in contributing a fix?

Reply:
Yes of course, I opened a pull request: https://github.com/microsoft/playwright/pull/28239.